docs/reference/reference.rst

```diff
@@ -2,6 +2,8 @@
 Package reference
 *****************
 
-.. toctree::
+.. autosummary::
+   :toctree:
+   :recursive:
 
-   gen/craft_parts
```

Comment:
Ah I see now why the new version looks cleaner; `gen/` has the result of apidoc generating an individual rst file from each python module, with the proper `automodule` directives. So with this change the documentation is actually missing a lot.
For example, compare the [current docs](https://canonical-craft-parts.readthedocs-hosted.com/en/latest/reference/gen/craft_parts.plugins.ant_plugin.html) for the ant module with the [new one](https://dboddie-test-org.github.io/craft-parts-docs/reference/craft_parts.plugins.ant_plugin.html#module-craft_parts.plugins.ant_plugin) (the one with these changes).

Reply:
Yes, that's missing a lot. I'll see if there's another way to keep the reference hierarchy while preventing warnings about duplicate documentation.